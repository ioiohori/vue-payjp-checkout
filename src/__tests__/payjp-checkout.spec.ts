import { shallow, Wrapper } from '@vue/test-utils';
import PayjpCheckout from '../';

describe('PayjpCheckout', () => {
  let propsData: () => object,
    component: () => Wrapper<PayjpCheckout>,
    global = window as any;
  beforeEach(() => {
    delete global.onCreatedPayjpToken_45cfff31a1607c000;
    delete global.onFailedPayjpToken_3befe4a56b4e64000;
    propsData = () => ({
      apiKey: 'asdfghjk',
      clientID: '1234',
      text: 'pay!',
      submitText: '"submit!',
      tokenName: 'test token',
      previousToken: 'foo',
      namePlaceholder: 'John Doe',
      createdCallbackName: 'onCreatedPayjpToken_45cfff31a1607c000',
      failedCallbackName: 'onFailedPayjpToken_3befe4a56b4e64000'
    });
    component = () =>
      shallow(PayjpCheckout, {
        propsData: propsData()
      });
  });

  it('matches snapshot', () => {
    expect(component().html()).toMatchSnapshot();
  });

  it('fills callback globally', () => {
    expect(global.onCreatedPayjpToken_45cfff31a1607c000).toBeUndefined();
    expect(global.onFailedPayjpToken_3befe4a56b4e64000).toBeUndefined();
    component();
    expect(global.onCreatedPayjpToken_45cfff31a1607c000).toBeInstanceOf(
      Function
    );
    expect(global.onFailedPayjpToken_3befe4a56b4e64000).toBeInstanceOf(
      Function
    );
    global.onCreatedPayjpToken_45cfff31a1607c000();
    expect(global.onCreatedPayjpToken_45cfff31a1607c000).toBeUndefined();
    global.onFailedPayjpToken_3befe4a56b4e64000();
    expect(global.onFailedPayjpToken_3befe4a56b4e64000).toBeUndefined();
  });
});
