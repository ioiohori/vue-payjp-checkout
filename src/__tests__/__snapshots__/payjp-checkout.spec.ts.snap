// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayjpCheckout matches snapshot 1`] = `"<div><script src=\\"https://checkout.pay.jp/\\" class=\\"payjp-button\\" data-key=\\"asdfghjk\\" data-payjp=\\"1234\\" data-partial=\\"false\\" data-text=\\"pay!\\" data-submit-text=\\"&quot;submit!\\" data-token-name=\\"test token\\" data-previous-token=\\"foo\\" data-lang=\\"ja\\" data-on-created=\\"onCreatedPayjpToken_45cfff31a1607c000\\" data-on-failed=\\"onFailedPayjpToken_3befe4a56b4e64000\\" data-name-placeholder=\\"John Doe\\"></script></div>"`;
